### You will work with a simple dummy data set that contains factory production values of electrical machinery parts for a corporation.

### **Task:**

To Calculate Production values as per the days and existing infrastructure?

##### You will perform the following tasks:
- Load and study the data
- Extract production values as a matrix from the data
- Extract number of working days as a matrix from the data
- Multiply matrices

In [ ]:
import numpy as np
import pandas as pd

In [ ]:
df = pd.read_csv('/content/Factory_Production.csv',index_col='Factory')
df

,Generators,Motors,Cables,Days
Factory,,,,
A,8,17,232,352
B,6,20,203,348
C,10,16,187,337
D,7,24,218,357
E,13,13,256,362
F,9,11,179,365
G,7,15,156,339
H,5,8,223,324
I,12,18,187,316


# **Matrix Multiplication**

![](https://www.mathsisfun.com/algebra/images/matrix-multiply-ex1b.svg)

#### Feature Description:
This data set contains the production values of power units such as generators, motors and cables of different factories belonging to the same corporation.

- Factory = unique label assigned to a factory unit for the purpose of identification
- Generators = number of generators produced by the factory in a day
- Motors = number of motors produced by the factory in a day
- Cables = number of cables produced by the factory in a day
- Days = number of working days of the factory in a year

##### Study its features such as:
- The number of factory units
- The number of machine parts
- The ranges of production values

## Task 2 - Extract production values as a matrix from the data

- The factory production values for each factory are contained in the "Generators", "Motors" and "Cables" features
- These production values will form a 10 x 3 matrix (considering all factories)

In [ ]:
df_prod = df.drop('Days',axis=1)
df_prod

,Generators,Motors,Cables
Factory,,,
A,8,17,232
B,6,20,203
C,10,16,187
D,7,24,218
E,13,13,256
F,9,11,179
G,7,15,156
H,5,8,223
I,12,18,187


## Task 3 - Extract number of working days as a matrix from the data

- The number of working days for each factory is stored in the "Days" column of the data frame "df"
- These values will form a 10 x 1 matrix

In [ ]:
df_days = df['Days']
df_days

Factory
A    352
B    348
C    337
D    357
E    362
F    365
G    339
H    324
I    316
J    346
Name: Days, dtype: int64

## Task 4 - Multiply matrices

- The factory production values of generators, motors and cables of the 10 factories is a 10 x 3 matrix
- The number of working days of all the factories is in the form of a 1 x 10 matrix
- The product of these two matrices is equal to the linear combination of the production values vectors scaled by the number of working days

In [ ]:
A = np.matrix(df_prod)
X = np.matrix(df_days)

In [ ]:
A.shape

(10, 3)

In [ ]:
X.shape

(1, 10)

## Task 5 - Code a linear combination of vectors using addition and scalar multiplication

- The vector of daily production values for a certain factory can be accessed from the data frame "df_prod".
- The number of working days in a year for each factory can be extracted from the data frame "df".
- The total production of a certain factory in a full year can be obtained using scalar multiplication.

![](https://www2.seas.gwu.edu/~simhaweb/lin/modules/module3/figures/matrixvec2.png)

In [ ]:
data = np.array(X*A).flatten()
pd.Series(data,index=df_prod.columns)

Generators     30355
Motors         56602
Cables        708048
dtype: int64

##### Observations

- The corporation produces 30355 generators in a year
- The corporation produces 56602 motors in a year
- The corporation produces 708048 cables in a year

### Conclusions

- From the factory production data, we can calculate the total production of the corporation
- We can treat the production values of all the factories as a matrix
- We can treat the number of working days of factories as a matrix
- The product of the two matrices is equal to the linear combination of the factory production values scaled by their number of working days
- We can multiply the two matrices to calculate the total production